api/envoy/config/filter/http/cache/v2/cache.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.cache.v2;
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.http.cache.v2";
+option java_outer_classname = "CacheProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filters.http.cache.v3alpha";
+
+// [#protodoc-title: HTTP Cache Filter]
+// HTTP Cache Filter :ref:`overview <config_http_filters_cache>`.
+// [#extension: envoy.filters.http.cache]
+message CacheConfig {
+  // List of query params.
+  message QueryParams {
+    repeated string query_string_keys = 1;
+  }
+
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
```

Comment:
The more I think about this, the more I think even these parts should look like route matchers.

Reply:
I switched to api.v2.route.QueryParameterMatcher and type.matcher.StringMatcher where they made sense, but I don't see any way to improve on the bools for exclude_scheme and exclude_host, or the uint32 for max_body_bytes.